pkg/util/annotation.go

```diff
@@ -29,3 +47,31 @@ func GetAnnotationValue(annotations map[string]string, annotationKey string) str
 	}
 	return annotations[annotationKey]
 }
+
+func getDeletedAnnotationKeys(desired, observed *unstructured.Unstructured) sets.Set[string] {
+	recordKeys := sets.NewString(strings.Split(observed.GetAnnotations()[workv1alpha2.ManagedAnnotation], ",")...)
```

Comment:
1. what if observed.GetAnnotations() == nil? This may not happen in normal cases, but who knows what people will do, like manually deleting all annotations directly from the member cluster.

2. Please don't use `sets.NewString` anymore because it has been deprecated now.

```go
func getDeletedAnnotationKeys(desired, observed *unstructured.Unstructured) sets.Set[string] {
	last := observed.GetAnnotations()
	if last == nil {
		return nil
	}

	lastKeys := sets.New[string](strings.Split(last[workv1alpha2.ManagedAnnotation], ",")...)

	cur := desired.GetAnnotations()
	if cur == nil {
		return lastKeys
	}

	for key := range desired.GetAnnotations() {
		lastKeys.Delete(key)
	}

	return lastKeys
}
```

Reply:
> what if observed.GetAnnotations() == nil?

https://go.dev/play/p/VLWyHAQ87rb